tests/metric.rs

```diff
@@ -0,0 +1,119 @@
+//! ## Integration test of `AWS IoT Device defender metrics`
+//!
+//!
+//! This test simulates publishing of metrics and expects a accepted response from aws
+//!
+//! The test runs through the following update sequence:
+//! 1. Setup metric state
+//! 2. Assert json format
+//! 2. Publish metric
+//! 3. Assert result from AWS
```

Comment:
This part does not seem to happen? Not sure what would be the easiest way here, without pulling in a full AWS SDK.

I think what you do is okay for this, but maybe just change this comment to reflect what is actually happening?